apps/web/src/components/energy/views/today-view.tsx

```diff
@@ -0,0 +1,38 @@
+import { AggregationType } from "@energyleaf/lib";
+import { Skeleton } from "@energyleaf/ui/skeleton";
+import { getTimezoneOffset } from "date-fns-tz";
+import { Suspense } from "react";
+import AbsoluteChartView from "../charts/absolute-chart-view";
+import HourChartView from "../charts/hour-chart-view";
+import PeaksView from "./peak-view";
+
+interface Props {
+    initialDate?: Date;
+}
+
+export default async function EnergyPageTodayView(props: Props) {
+    const offset = getTimezoneOffset("Europe/Berlin", props.initialDate ? props.initialDate : new Date());
+    const localOffset = Math.abs(new Date().getTimezoneOffset() * 60 * 1000);
+
+    const serverStartDate = props.initialDate ? new Date(props.initialDate) : new Date();
+    serverStartDate.setHours(0, 0, 0, 0);
+    const serverEndDate = props.initialDate ? new Date(props.initialDate) : new Date();
+    serverEndDate.setHours(23, 59, 59, 999);
+
+    const startDate = offset === localOffset ? serverStartDate : new Date(serverStartDate.getTime() - offset);
```

Comment:
Kann man das Ermitteln des korrekten Datums vllt. noch in eine Methode auslagern? Das machst du ja an mehreren Stellen, dann kann man das später evtl. einfacher anpassen